### ISSUE TITLE
pprint-xml should use shortened syntax when a tag has no children

### ISSUE THREAD

**Issue by hyphz** [slavapestov/factor#1287]
At the moment, if a tag's children are an empty list, pprint-xml prints it like:
    \<tag attrs="attrs"> \</tag>
It would be nice if it could use the shortened XML syntax for this case:
    \<tag attrs="attrs" />

**Comment by mrjbq7** [comment]
This fixes it -- seems to be a problem with having an empty sequence (e.g., ``{ }``) as children, the predicate class was only checking for ``f`` or not (boolean-ness).

```diff
diff --git a/basis/xml/data/data.factor b/basis/xml/data/data.factor
index 71d5946..27f2d5a 100644
--- a/basis/xml/data/data.factor
+++ b/basis/xml/data/data.factor
@@ -214,8 +214,8 @@ M: xml like
 : <contained-tag> ( name attrs -- tag )
     f <tag> ;
 
-PREDICATE: contained-tag < tag children>> not ;
-PREDICATE: open-tag < tag children>> ;
+PREDICATE: contained-tag < tag children>> empty? ;
+PREDICATE: open-tag < tag children>> empty? not ;
 
 TUPLE: unescaped string ;
 C: <unescaped> unescaped
```

@erg, thoughts?

**Comment by erg** [comment]
Looks good!

**Comment by mrjbq7** [comment]
Fixed in c7f440e097f94abca26fd474ffac1d78f6fe4d97.
